site/app/libraries/database/DatabaseQueries.php

```diff
@@ -5171,6 +5186,35 @@ public function getInstructorLevelAccessCourse(string $user_id): array {
         return $this->submitty_db->rows();
     }
 
+    /**
+     * @return array<Course>
+     */
+    public function getSelfRegisterCourses(string $user_id): array {
+        $query = <<<SQL
+SELECT c.*, t.name AS term_name FROM courses c, terms t
+WHERE NOT c.self_registration_type = 0 AND c.status = 1 and c.course NOT IN (
```

Comment:
It might be better to do this for clarity about the order of operations:

```suggestion
WHERE c.self_registration_type > 0 AND c.status = 1 and c.course NOT IN (
```

You could also directly inject the class constants in this string so the numbers are less magical.

Reply:
This has been done (for self registration type)